File: tests/fixtures/edgecase_repo/csharp/Service4.cs
using System;

namespace Example.Gen
{
    public interface IHandler4
    {
        bool Handle(string payload);
    }

    public class Base4
    {
        protected string Name;
    }

    public class Service4 : Base4, IHandler4
    {
        public Service4(string name)
        {
            this.Name = name;
        }

        public bool Handle(string payload)
        {
            this.Validate(payload);
            return true;
        }

        private void Validate(string payload) { }
    }
}

File: tests/fixtures/edgecase_repo/kotlin/Service4.kt
package com.example.gen

import com.example.gen.Base4

interface Handler4 {
    fun handle(payload: String): Boolean
}

open class Base4(val name: String)

class Service4(name: String) : Base4(name), Handler4 {
    override fun handle(payload: String): Boolean {
        validate(payload)
        return true
    }

    private fun validate(payload: String) {}
}

fun helper4(value: String): String {
    return value
}
